src/efs_parser/BigWig.py

```diff
@@ -184,7 +184,15 @@ def getRange(self, chr, start, end, bins=2000, zoomlvl=-1, metric="AVG", respTyp
             if respType == "DataFrame":
                 result = toDataFrame(values, self.columns)
                 result["chr"] = chr
+                # checking range by: Hany Elgaml
 
+                query_string = '(start >= ' + str(start) + ' and start <= ' + str(end) + ') or (end >= ' + str(
+                    start) + ' and end <= ' + str(end) + ')'
+                result = result.query(query_string)
+                result.insert(1, "new", range(0, len(result)))
+                result.set_index('new', inplace=True)
```

Comment:
we don't need to reindex the result, so lets remove these lines

Reply:
done